csi/moac/pvhandler.ts

```diff
@@ -0,0 +1,74 @@
+const log = require('./logger').Logger('pvhandler');
+
+import {
+  CoreV1Api,
+  KubeConfig,
+  V1PersistentVolume,
+  Informer,
+  makeInformer,
+} from 'client-node-fixed-watcher'
+import { Volumes } from './volumes';
+
+
+const kc = new KubeConfig();
+kc.loadFromDefault();
+const k8sApi = kc.makeApiClient(CoreV1Api);
+const listFn = () => k8sApi.listPersistentVolume();
+
+export class PVHandler {
+  informer: Informer<V1PersistentVolume>
+  listPvFn = k8sApi.listPersistentVolume();
+  volumes: Volumes; // Volume manager
+
+  constructor(
+    volumes: Volumes,
+  ) {
+    this.volumes = volumes
+    log.debug(`PVHandler, created`);
+    this.informer = makeInformer(kc, '/api/v1/persistentvolumes', listFn)
+    this.informer.on('delete', async (obj: V1PersistentVolume) => { 
+      if (obj.spec?.csi?.driver === 'io.openebs.csi-mayastor') {
+        // Initiate deletion of the MSV here.
+        // Deletion of an already deleted MSV should not fail,
+        // it may have been deleted already if reclaimPolicy is not Retain.
+        // FIXME: We need to avoid a race between 2 paths attempting to delete
+        // the associated MSV,
+        // Possible fixes:
+        //  1. Single path, i.e. delegate MSV deletion to this informer only.
+        //  2. "Deleting" an MSV => pushing an event on a queue,
+        //      the semantics of the event are "delete identified MSV if found"
+        //  3. Generate the 'del' event for the volume, this has the desired 
+        //    semantics see VolumeOperator._bindWatcher
+        // Simplest solution is 3 and the one selected.
```

Comment:
I'm not sure that solution #2 is difficult either and is less error-prone. IMO the only thing that is needed is to pass volumeOperator reference to PvOperator constructor and call `_deleteResource` method that must be made public (so `deleteResource`). And you don't need to pass volumes object reference to PvOperator then and it's cleaner and easier to understand:

1. del volume object event retains its original meaning - the volume has been destroyed (and not the volume *should* be destroyed in a near future).
2. it is aligned with the overall architecture. The whole thing that PvOperator does is that it automates the step of deleting msv when pv is gone, that would otherwise have to be done manually. Logically from that follows that the operator will be using msv operator to do exactly what would have been done otherwise manually.

Reply:
I tried #2 but it result in left over MSVs. 
I've elected to simplify, the pv operator now invokes `volumes.destroyVolume` which has the desired semantics.
This does mean moac emits a few more warning messages but is far more simpler and less contorversial.